src/core/internalMemMgr.cpp

```diff
@@ -563,4 +576,116 @@ uint32 InternalMemMgr::GetReferencesCount()
     return static_cast<uint32>(m_references.NumElements());
 }
 
+Result InternalMemMgr::Map(
+        GpuMemory* pGpuMemory,
+        void** ppData)
+{
+    Util::MutexAuto allocatorLock(&m_allocatorLock); // Ensure thread-safety using the lock
```

Comment:
Calling `WasBuddyAllocated()` is fine to do outside the lock, so the lock should only be taken inside the scope of that check.

Reply:
Done